txscript/engine.go

```diff
@@ -555,6 +560,13 @@ func (vm *Engine) checkHashTypeEncoding(hashType SigHashType) error {
 // checkPubKeyEncoding returns whether or not the passed public key adheres to
 // the strict encoding requirements if enabled.
 func (vm *Engine) checkPubKeyEncoding(pubKey []byte) error {
+	if vm.hasFlag(ScriptVerifyWitnessPubKeyType) && vm.witness &&
```

Comment:
Disaster averted! Ran with your suggestion and guarded all the segwit behavior implemented in this commit behind a check of a `vm.isWitnessVersionActive(0)`. Upon re-reviewing the relevant BIPs, this should've been done the first time around as all the BIPs only specify the defined rules for a `v0` witness program.